UIExtenderLibModule/ViewModel/Patches/GauntletPartyScreenCallsites.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using HarmonyLib;
+using SandBox.GauntletUI;
+using TaleWorlds.Core;
+
+namespace UIExtenderLibModule.ViewModel.Patches
+{
+    //  Harmony cannot resolve the interface methods by iself, so this route has to be used
+    [HarmonyPatch]
+    public static class GauntletPartyScreenOnActivateCallsite
+    {
+        public static MethodBase TargetMethod()
+        {
+            Type type = typeof(GauntletPartyScreen);
+            MethodInfo interfaceMethod = type.GetInterface(nameof(IGameStateListener)).GetMethod("OnActivate");
```

Comment:
There is also an `OnInitialize` method that may be a better point to use as it is most likely called once instead of multiple times. As far as I understand the library as of now this only needs to be done once.

Reply:
No, OnInitialize does not fire correctly for the first time the party menu is opened. I have already explored this. It does fire each subsequent time the menu is opened. I can add the hook here for it back in, but for my purposes the various sorting functions occur on button presses, rather than on menu open